library/ssl_cookie.c

```diff
@@ -151,27 +151,74 @@ int mbedtls_ssl_cookie_setup( mbedtls_ssl_cookie_ctx *ctx,
 /*
  * Generate the HMAC part of a cookie
  */
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+static int ssl_cookie_hmac( mbedtls_ssl_cookie_ctx *ctx,
```

Comment:
I think the code would be a bit simpler if we had a single prototype - `mbedtls_ssl_cookie_ctx *ctx` should work in both cases. This simplifies call sites and reduces `#ifdef` fatigue by concentrating them in a one place.

Reply:
Oh, I just saw again your note on the issue:

> I think the `ssl_cookie_hmac()` should be non-PSA only and an inline `psa_mac_sign_XX` code should replace the only remaining call to `ssl_cookie_hmac()`

If you do that change (which I agree looks good), then my point above is moot.

Reply:
I wonder if I should rewrite history at this point ? Like directly write in-line PSA code instead of calling ssl_cookie_hmac

Reply:
I have a slight preference for not rewriting history in this case.